creator-node/test/stateMonitoringUtils.test.js

```diff
@@ -876,3 +880,193 @@ describe('test aggregateReconfigAndPotentialSyncOps()', function () {
+      primaryClock = 10
+      secondaryClock = 5
+      primaryFilesHash = '0x123'
+      secondaryFilesHash = '0x456'
+
+      // Mock DBManager.fetchFilesHashFromDB() to return `secondaryFilesHash` for clock range
+      const DBManagerMock = DBManager
+      DBManagerMock.fetchFilesHashFromDB = async () => {
+        return secondaryFilesHash
+      }
+      proxyquire('../src/services/stateMachineManager/stateMonitoring/stateMonitoringUtils', {
+        '../../../dbManager': DBManagerMock
+      })
+
+      const syncMode = await computeSyncModeForUserAndReplica({
+        wallet,
+        primaryClock,
+        secondaryClock,
+        primaryFilesHash,
+        secondaryFilesHash
+      })
+
+      assert.strictEqual(syncMode, SYNC_MODES.SyncSecondaryFromPrimary)
+    })
+
+    it('Returns SYNC_MODES.SyncPrimaryFromSecondary if primaryFilesHashForRange != secondaryFilesHash', async function () {
+      primaryClock = 10
+      secondaryClock = 5
+      primaryFilesHash = '0x123'
+      secondaryFilesHash = '0x456'
+      primaryFilesHashMock = '0x789'
+
+      // Mock DBManager.fetchFilesHashFromDB() to return different filesHash for clock range
+      const DBManagerMock = DBManager
+      DBManagerMock.fetchFilesHashFromDB = async () => {
+        return primaryFilesHashMock
+      }
+      proxyquire('../src/services/stateMachineManager/stateMonitoring/stateMonitoringUtils', {
+        '../../../dbManager': DBManagerMock
+      })
+
+      const syncMode = await computeSyncModeForUserAndReplica({
+        wallet,
+        primaryClock,
+        secondaryClock,
+        primaryFilesHash,
+        secondaryFilesHash
+      })
+
+      assert.strictEqual(syncMode, SYNC_MODES.SyncPrimaryFromSecondary)
+    })
+
+    it("Throws error primaryFilesHashForRange can't be retrieved", async function () {
+      // Increase mocha test timeout from default 2s to accommodate `async-retry` runtime
+      this.timeout(30000) // 30s
+
+      primaryClock = 10
+      secondaryClock = 5
+      primaryFilesHash = '0x123'
+      secondaryFilesHash = '0x456'
+
+      // Mock DBManager.fetchFilesHashFromDB() to throw error
+      const DBManagerMock = require('../src/dbManager')
+      DBManagerMock.fetchFilesHashFromDB = async () => {
+        throw new Error('Mock - Failed to fetch filesHash')
+      }
+      proxyquire('../src/services/stateMachineManager/stateMonitoring/stateMonitoringUtils', {
+        '../../../dbManager': DBManagerMock
+      })
+
+      try {
+        await computeSyncModeForUserAndReplica({
+          wallet,
+          primaryClock,
+          secondaryClock,
+          primaryFilesHash,
+          secondaryFilesHash
+        })
+      } catch (e) {
+        assert.strictEqual(
+          e.message,
+          '[computeSyncModeForUserAndReplica] Error: failed DBManager.fetchFilesHashFromDB() - Mock - Failed to fetch filesHash'
+        )
+      }
+    })
+  })
+})
```

Comment:
this looks very comprehensive but I'm curious if it shows any lines missing in `computeSyncModeForUserAndReplica()` when you run `npm run test:coverage` ?

Reply:
good call, i made a change to always log code coverage in CI (see below commits' CI)

as you can see, the only uncovered lines are 475-477, [which are here](https://github.com/AudiusProject/audius-protocol/pull/3310/files#diff-dc51c22783493f061b9453d096c1a6a01091b33bc2686cd0e71d33c574cf9efeR475), but seems like those are actually covered by the [final test case here](https://github.com/AudiusProject/audius-protocol/pull/3310/files#diff-d6306b5bad7be2812926a9457bfa1f9c22d8a359590a35dcfc3f3fe79a145f8fR1038)?
do i need to do smth additional or differently?